src/local_scheduler/local_scheduler.cc

```diff
@@ -130,6 +130,19 @@ void kill_worker(LocalSchedulerState *state,
     /* Update the task table to reflect that the task failed to complete. */
     if (state->db != NULL) {
       Task_set_state(worker->task_in_progress, TASK_STATUS_LOST);
+      /* Insert a dummy object to notify waiters of the task failure. */
+      TaskSpec *spec = Task_task_spec(worker->task_in_progress);
+      int64_t num_returns = TaskSpec_num_returns(spec);
+      for (int i=0; i < num_returns; i++) {
+        ObjectID object_id = TaskSpec_return(spec, i);
+        uint8_t *data;
+        Status status = state->plasma_conn->Create(object_id.to_plasma_id(), 1, NULL, 0, &data);
+        if (status.ok()) {
+          ARROW_CHECK_OK(state->plasma_conn->Seal(object_id.to_plasma_id()));
+        } else {
+          CHECK(status.IsPlasmaObjectExists());
+        }
+      }
```

Comment:
It probably isn't quite sufficient to just create an object for the currently exiting task. We also need to do it for all queued tasks (if it's an actor) and all tasks that are assigned to the actor in the future.

Reply:
Yikes you're right, that makes this a lot more complicated.